### PR TITLE
Add locale number support to InputComponent

### PR DESCRIPTION
**Pull request by colincarter**
This swaps commas and dots to provide a parsable value while retaining the typed in number for display.

In addition InputComponent has been converted to typescript and cleaned up a bit.

PR used in conjunction with https://github.com/axsy-dev/react-app/pull/7165

### CODE REVIEW

File: src/fields/InputField.ios.js [hunk 1]
```diff
@@ -1,39 +1,38 @@
-'use strict';
+"use strict";
```
Comment (colincarter) [inline src/fields/InputField.ios.js:1]: prettier formatting on this file

File: src/lib/InputComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,280 @@
+import React from "react";
```
Comment (colincarter) [inline src/lib/InputComponent.tsx:1]: This file was converted to typescript from `InputComponent.tsx`.  I've attempted to improve typing in this file, but it's still not complete

File: src/lib/InputComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+type ValidationFunction = (
+  value: State["value"],
+  scope: InputComponent
+) => boolean;
+
+type Props = TextInputProps & {
+  height: number;
+  containerStyle: Record<string, any>;
+  isDecimal: boolean;
+  numberFormats: { decimalSeparator: string; groupingSeparator: string };
```
Comment (colincarter) [inline src/lib/InputComponent.tsx:23]: These `numberFormats` come react-native-localize in react-app

File: src/lib/InputComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+      const localedValues = this.handleLocale(value);
+      value = localedValues.converted;
+      displayValue = localedValues.display;
+    }
+
+    this.validate(value);
+
+    this.setState({
+      value,
+      displayValue,
```
Comment (colincarter) [inline src/lib/InputComponent.tsx:198]: displayValue from state is used as the raw value the user types in.

File: src/lib/InputComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+    if (this.props.isDecimal) {
+      const localedValues = this.handleLocale(value);
+      value = localedValues.converted;
+      displayValue = localedValues.display;
+    }
+
+    this.validate(value);
+
+    this.setState({
+      value,
```
Comment (colincarter) [inline src/lib/InputComponent.tsx:197]: value is the converted displayValue

File: src/lib/InputComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+      // e.g. for a european number: 120.000,99 would become 120,000.99
+      converted = value
+        .replace(groupingSeparator, "___")
+        .replace(decimalSeparator, groupingSeparator)
+        .replace("___", decimalSeparator);
+    }
+
+    return {
+      display: value,
+      converted: converted.replace(DefaultNumberFormats.groupingSeparator, "") // commas to form a valid parsable number
```
Comment (glenn-axsy) [inline src/lib/InputComponent.tsx:228]: A bit nit-picky, but the grouping separator can also be "." or " ". If you just remove the word "commas" this comment would make more general sense.

File: src/lib/InputComponent.tsx (hunk 2 above)
Comment (glenn-axsy) [reply src/lib/InputComponent.tsx:1]: I think you meant it was converted from `InputComponent.js`.
I did a side-by side eyeball comparison and the conversion looks good.

**Review by glenn-axsy — APPROVED** [review]
Looks like a big improvement. I just had one nit-pick about a comment that assumes that the comma will be the grouping separator.

File: src/lib/InputComponent.tsx (hunk 6 above)
Comment (colincarter) [reply src/lib/InputComponent.tsx]: Fixed in https://github.com/axsy-dev/react-native-form-generator/pull/14/commits/44043b1c6494b356f8adae3775a8f67f277c2a5e

**Pull request merged**